src/tracer.js

```diff
@@ -0,0 +1,258 @@
+        this.registerExtractor(opentracing.FORMAT_HTTP_HEADERS, httpCodec);
+    }
+
+    _startInternalSpan(
+            spanContext: SpanContext,
+            operationName: string,
+            startTime: number,
+            tags: any = {},
+            rpcServer: boolean): Span {
+
+        let firstInProcess = rpcServer || (spanContext.parentId == null);
+        let span = new Span(
+            this,
+            operationName,
+            spanContext,
+            startTime,
+            firstInProcess
+        );
+
+        if (opentracing_tags.SAMPLING_PRIORITY in tags) {
+            span.setSamplingPriority(tags[opentracing_tags.SAMPLING_PRIORITY])
+        }
+
+        span.addTags(tags);
+        return span;
+    }
+
+    _report(span: Span): void {
+        span.addTags(this._tracerTags);
+        this._reporter.report(span);
+    }
+
+    registerInjector(format: string, injector: Injector): void {
+        this._injectors[format] = injector;
+    }
+
+    registerExtractor(format: string, extractor: Extractor): void {
+        this._extractors[format] = extractor;
+    }
+
+    /**
+    * The method for creating a root or child span.
+    *
+    * @param {object} fields - the fields to set on the newly created span.
+    * @param {string} fields.operationName - the name to use for the newly
+    *        created span. Required if called with a single argument.
+    * @param {SpanContext} [fields.childOf] - a parent SpanContext (or Span,
+    *        for convenience) that the newly-started span will be the child of
+    *        (per REFERENCE_CHILD_OF). If specified, `fields.references` must
+    *        be unspecified.
+    * @param {array} [fields.references] - an array of Reference instances,
+    *        each pointing to a causal parent SpanContext. If specified,
+    *        `fields.childOf` must be unspecified.
+    * @param {object} [fields.tags] - set of key-value pairs which will be set
+    *        as tags on the newly created Span. Ownership of the object is
+    *        passed to the created span for efficiency reasons (the caller
+    *        should not modify this object after calling startSpan).
+    * @param {number} [fields.startTime] - a manually specified start time for
+    *        the created Span object. The time should be specified in
+    *        milliseconds as Unix timestamp. Decimal value are supported
+    *        to represent time values with sub-millisecond accuracy.
+    * @return {Span} - a new Span object.
+    **/
+    startSpan(fields: spanFields): Span {
+        let tags = fields.tags || {}
+        let references = fields.references || [];
+        let startTime = fields.startTime;
+        if (!startTime) {
+            startTime = Utils.getTimestampMicros();
+        }
+
+        let parent: ?SpanContext;
+        for (let i = 0; i < references.length; i++) {
+            let ref: Reference = references[i];
+            if (ref.type() === opentracing.REFERENCE_CHILD_OF) {
```

Comment:
jaeger-go only references child_of, should it also be changed to accept follows_from?